summoner-tui/src/Summoner/Tui/Validation.hs

```diff
@@ -48,6 +49,15 @@ handleAutofill f =
           setFormFocus CustomPreludeName $ mkForm new_state
     _ -> f
 
+-- | Adds a newline for text fields.
+newLine :: KitForm e -> KitForm e
+newLine =
+  handleNewLine ProjectDesc (project . desc)
+  where
+    handleNewLine formField fieldLens f =
```

Comment:
I guess there is no need to introduce function here, you can inline it in the `newLine` function :slightly_smiling_face: But I don't mind having it, however, by our style guide the type should be written as well.

Reply:
I've added a type signature now.

I didn't want to inline it because I want this function to be applicable to other fields in the future. This way it can be done just by composing different newline handlers on `KitForm e`.

Reply:
Thanks for thinking about that. But currently, we don't have any other multiline or any big-enough fields for that, and nothing we are planning in the nearest future. Do you have any particular thoughts on that?

Reply:
> nothing we are planning in the nearest future. Do you have any particular thoughts on that?

I would still prefer having a generic "new line handler" function but if there is no such plans I'm okay with converting it to handle only description. I'm also changing function's name since it's now specific to project description.